.github/workflows/test.yml

```diff
@@ -3,6 +3,11 @@ on:
   schedule:
     - cron: "0 12 * * 1"
   push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```

Comment:
I'm wondering if this won't cover the scenario where you are developing on a branch (not main, typically) on your fork and you want to just test there before doing a PR. Anyway, I'm fine with this, just keep an eye on whether tests are being run twice, and if the need arises to run tests on other branches.